docker-compose.yml

```diff
@@ -0,0 +1,33 @@
+services:
+  backend:
+    build: ./backend
+    container_name: express-backend
+    ports:
+      - "5001:5001"
+    networks:
+      - matcha
+
+  frontend:
+    build: ./frontend
+    container_name: vite-frontend
+    ports:
+      - "5173:5173"
+    networks:
+      - matcha
+
+  nginx:
+    image: nginx:alpine
+    container_name: nginx-gateway
+    ports:
+      - "80:80"
+    volumes:
+      - ./nginx/nginx.conf:/etc/nginx/nginx.conf:ro
```

Comment:
The configuration file could be store in `/etc/nginx/conf.d` to avoid re-configuring the `events` and `http` block.